draft-ietf-httpbis-message-signatures.md

```diff
@@ -287,7 +287,7 @@ The following sections define component identifier names, their parameters, thei
 
 The component name for an HTTP field is the lowercased form of its field name as defined in {{Section 5.1 of HTTP}}. While HTTP field names are case-insensitive, implementations MUST use lowercased field names (e.g., `content-type`, `date`, `etag`) when using them as component names.
 
-The component value for an HTTP field is the field value for the named field as defined in {{Section 5.5 of HTTP}}. The field value MUST be taken from the named header field of the target message unless this behavior is overridden by additional parameters and rules, such as the `req` and `tr` flags, below.
+The component value for an HTTP field is the field value for the named field as defined in {{Section 5.5 of HTTP}}. The field value MUST be taken from the named header field of the target message unless this behavior is overridden by additional parameters and rules, such as the `req` and `tr` flags, below. For most fields, the field value is an ASCII string. Other encodings could exist, but extra care needs to be taken when dealing with these values to prevent inclusion if illegal octets (such as a newline character).
```

Comment:
I don't think "extra care" is sufficient here. If the field value contains an octet >= 128, it can't eb placed into the signature base, which is said use ASCII. The spec needs to exactly tell implementers what to do here.

Reply:
What would you suggest? The obvious thing is that if an octet is not in the ASCII range then it needs to be encoded, likely percent-encoded first.

Reply:
percent encoding works for me, but you need to make sure that then "%" is percent-escaped as well.

Reply:
I still don't think "extra care" is sufficient here. Just say what the implementer needs to do.